### ISSUE TITLE
Unify with in-tree compiletest-rs

### ISSUE THREAD

**Issue by Manishearth** [Manishearth/compiletest-rs#238]
I'd like for us to eventually have a world where rustc simply depends on compiletest-rs, and compiletest-rs lives in rust-lang

Blockers:
 - [ ] Remove dependency on `test` crate https://github.com/Manishearth/compiletest-rs/issues/237

**Comment by Munksgaard** [comment]
I sorta-kinda tried to address this in #201, but never got around to finishing it. My impression was that rustc had some exotic requirements for its internal compiletest that were hard to generalise for this package. 

I don't expect I'll get back to working on that PR anytime soon, and it's probably diverged too far from upstream by now to be of any real use, but it is there, if anyone wants to take a look.

**Comment by Manishearth** [comment]
Yeah I'm wondering if we can keep those requirements but perhaps behind a flag.

**Comment by Manishearth** [comment]
Instead of doing a merge, we could potentially try a "fresh start" where we _first_ try moving compiletest out of tree, and if that works, release it as a breaking version bump to this crate, making sure major users (clippy, syn) have everything they need to upgrade.

**Comment by Munksgaard** [comment]
> I really _don't_ want us to have to maintain this as a fork smile

Yeah, we should try to minimize the maintenance burden as much as possible.

**Comment by Munksgaard** [comment]
> Instead of doing a merge, we could potentially try a "fresh start" where we _first_ try moving compiletest out of tree, and if that works, release it as a breaking version bump to this crate, making sure major users (clippy, syn) have everything they need to upgrade.

That may not be a bad idea. At least, it's worth a shot :-)

**Comment by Manishearth** [comment]
Filed rustc side of this issue in https://github.com/rust-lang/rust/issues/82946
